secretstores/azure/keyvault/authutils.go

```diff
@@ -69,7 +93,65 @@ func (c ClientAuthorizer) ServicePrincipalTokenByCertBytes() (*adal.ServicePrinc
 	return adal.NewServicePrincipalTokenFromCertificate(*oauthConfig, c.ClientID, certificate, rsaPrivateKey, c.Resource)
 }
 
-func (c ClientAuthorizer) decodePkcs12(pkcs []byte, password string) (*x509.Certificate, *rsa.PrivateKey, error) {
+// MSIConfig provides the options to get a bearer authorizer through MSI.
+type MSIConfig struct {
+	Resource string
+	ClientID string
+}
+
+// NewMSIConfig creates an MSIConfig object configured to obtain an Authorizer through MSI.
+func NewMSIConfig() MSIConfig {
+	return MSIConfig{
+		Resource: azure.PublicCloud.ResourceManagerEndpoint,
+	}
+}
+
+// GetMSI creates a MSI config object from the available client ID.
+func (k *keyvaultSecretStore) GetMSI(settings EnvironmentSettings) MSIConfig {
```

Comment:
same comments as earlier - avoid coupling.

Reply:
Resolved in fa4a4a0